modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -165,6 +166,37 @@ const cleanFormatNode = (editor: Editor, caretContainer: Node, formatNode: Eleme
   }
 };
 
+const normalizeNbsps = (node: SugarElement<Text>) => SugarText.set(node, SugarText.get(node).replace(new RegExp(`${Unicode.nbsp}$`), ' '));
+
+const normalizeNbspsBetween = (editor: Editor, caretContainer: Node | null) => {
+  const handler = () => {
+    if (caretContainer !== null && !editor.dom.isEmpty(caretContainer)) {
+      Traverse.prevSibling(SugarElement.fromDom(caretContainer)).each((node) => {
+        if (SugarNode.isText(node)) {
+          normalizeNbsps(node);
+        } else {
+          PredicateFind.descendant(node, (e) => SugarNode.isText(e)).each((textNode) => {
+            if (SugarNode.isText(textNode)) {
+              normalizeNbsps(textNode);
+            }
+          });
+        }
```

Comment:
This is the code that removes every nbsp in the previous sibling, and is what I was concerned about in the other thread.

Reply:
but `normalizeNbsps` removes only the last `&nbsp;`, it use this replace: `.replace(new RegExp(`${Unicode.nbsp}$`), ' ')`

`'a\u00a0b\u00a0\u00a0'.replace(new RegExp(`${Unicode.nbsp}$`), ' ')` -> `'a\u00a0b\u00a0 '`

Reply:
Oh _that_ was what you did in the latest refactor. Sorry, github makes it really hard to see what changed between looking at a review.